android/src/main/java/com/react/SmsModule.java

```diff
@@ -73,32 +73,33 @@ public void list(String filter, final Callback errorCallback, final Callback suc
             int ftid = filterJ.has("thread_id") ? filterJ.optInt("thread_id") : -1;
             String faddress = filterJ.optString("address");
             String fcontent = filterJ.optString("body");
+            String fContentRegex = filterJ.optString("bodyRegex");
             int indexFrom = filterJ.has("indexFrom") ? filterJ.optInt("indexFrom") : 0;
             int maxCount = filterJ.has("maxCount") ? filterJ.optInt("maxCount") : -1;
             String selection = filterJ.has("selection") ? filterJ.optString("selection") : "";
             String sortOrder = filterJ.has("sortOrder") ? filterJ.optString("sortOrder") : null;
             long maxDate = filterJ.has("maxDate") ? filterJ.optLong("maxDate") : -1;
             long minDate = filterJ.has("minDate") ? filterJ.optLong("minDate") : -1;
             Cursor cursor = context.getContentResolver().query(Uri.parse("content://sms/" + uri_filter), null, selection, null,
-                   	sortOrder);
+                    sortOrder);
             int c = 0;
             JSONArray jsons = new JSONArray();
 
             while (cursor != null && cursor.moveToNext()) {
-                boolean matchFilter = false;
+                boolean matchFilter = true;
                 if (fid > -1)
                     matchFilter = fid == cursor.getInt(cursor.getColumnIndex("_id"));
                 else if (ftid > -1)
                     matchFilter = ftid == cursor.getInt(cursor.getColumnIndex("thread_id"));
                 else if (fread > -1)
                     matchFilter = fread == cursor.getInt(cursor.getColumnIndex("read"));
-                else if ( faddress != null && !faddress.isEmpty() )
+                else if (faddress != null && !faddress.isEmpty())
                     matchFilter = faddress.equals(cursor.getString(cursor.getColumnIndex("address")).trim());
-                else if ( fcontent != null && !fcontent.isEmpty() )
+                else if (fcontent != null && !fcontent.isEmpty())
                     matchFilter = fcontent.equals(cursor.getString(cursor.getColumnIndex("body")).trim());
-                else {
-                    matchFilter = true;
-                }
+
+                if (fContentRegex != null)
```

Comment:
Usually I think it's better to error directly or print some error log on weird input so that we can find bugs early.

But for consistency, I agree to comply to the no-err philosophy, since other filters are also error robust.

Reply:
`"some string".matches("")` will always be false. I'll add the protection 👍